torch/fx/examples/replace_op.py

```diff
@@ -0,0 +1,58 @@
+import torch
+from torch.fx import symbolic_trace
+import re
+
+
+"""
+How to replace one op with another
+1. Iterate through all nodes in your GraphModule's Graph.
+2. Determine if the current node should be replaced. (Suggested: match
+on the node's ``target`` attribute).
+3. Create a replacement node.
+4. Use the fx built-in ``replace_all_uses_with`` to replace the current
+node with the replacement.
+5. Call ``recompile`` on the GraphModule. This updates the generated
+Python code to reflect the new Graph state.
+
+The following code demonstrates an example of replacing any instance of
+addition with a bitwise AND.
+"""
+
+# Sample module
+class M(torch.nn.Module):
+    def forward(self, x):
+        val = torch.neg(x) + torch.relu(x)
+        return torch.add(val, val)
+
+# Symbolically trace an instance of the module
+traced = symbolic_trace(M())
+
+# If you have `torch` imported, you have two ways to denonte addition:
+# the `+` operator and the method `torch.add`. `+` is a Python
+# built-in, so it's represented as a `call_function` Node with a target
+# of `<built-in function add>`. `torch.add` is part of `torch`, so
+# it becomes a `call_method` Node with a target of
+# `<built-in method add of type object at MEMORY-LOCATION-OF-TORCH>`.
+#
+# To determine whether a given node represents addition, we can match
+# on the `target` attribute. In this particular case, we have two
+# different representations of addition, so we'll use a regex to match
+# on `target`. (If you want to replace a `torch`-specific operator,
+# you can match on a simple string.)
```

Comment:
I think the things we should be matching on are a bit different, e.g.:

```
import torch
import torch.fx

class Additions(torch.nn.Module):
    def forward(self, x, y):
        return x + y, torch.add(x, y), x.add(y)

traced = torch.fx.symbolic_trace(Additions())
print(traced.graph)
"""
graph(x, y):
    %add_1 : [#users=1] = call_function[target=<built-in function add>](args = (%x, %y), kwargs = {})
    %add_2 : [#users=1] = call_function[target=<built-in method add of type object at 0x7f860323f7a0>](args = (%x, %y), kwargs = {})
    %add_3 : [#users=1] = call_method[target=add](args = (%x, %y), kwargs = {})
    return (add_1, add_2, add_3)
"""
```

the cases are:
1. call_function `<built-in function add>`. This is actually `operator.add` so we can match on equality with that function directly
2. call_function `<built-in method add of type object at 0x7f860323f7a0>` like you mentioned this is `torch.add`, which we can similarly match directly
3. This is the Tensor method call, so we can match target as a string

I think it's best do to:

```
patterns = [
    ('call_function', operator.add),
    ('call_function', torch.add),
    ('call_method', 'add')
]
```

Then later we can match like:

```
n : Node = ...
to_match = (n.op, n.target)
if any(to_match == pattern for pattern in patterns):
  # is a match, can swap stuff out
```

This should be safer than doing a string regex match.

BTW the operator form actually has an additional level of dispatch in the Python interpreter that goes through the string cat/addition and `__add__`, `__radd__` dispatch chain. It might be worth mentioning that the user should make sure that both operands are supposed to be tensors. (We don't have a way to confirm that directly, it's part of my operator normalization proposal doc)